### PR TITLE
Move some solar eclipse computations from AstroCalcDialog

### PR DESCRIPTION
**Pull request by 10110111**
This reduces the size of AstroCalcDialog.cpp by 15% while localizing much of solar eclipse map generation and some other related computations in another file.

### CODE REVIEW

**Review by alex-w — CHANGES_REQUESTED** [review]
Good improvement, thanks!

But please add new file (`src/core/SolarEclipseComputer.cpp`) into `po/stellarium/POTFILES.in` also and remove extra empty lines.

**Comment by 10110111** [comment]
The empty line at the end? Or is there some more?

**Comment by alex-w** [comment]
> The empty line at the end? Or is there some more?

See CodeFactor report please

**Review by alex-w — APPROVED** [review]

File: src/core/SolarEclipseComputer.hpp
Comment (gzotti) [inline src/core/SolarEclipseComputer.hpp:5]: Maybe reduce Nick? Or is some of his content from AstroCalc in this file?

Comment (alex-w) [reply src/core/SolarEclipseComputer.hpp:5]: +1 - Yes, the new file doesn't have Nick's code

Comment (alex-w) [reply src/core/SolarEclipseComputer.hpp:5]: Hmm... my name should be removed from new files too

**Review by gzotti — APPROVED** [review]
Yes, good idea to split this huge class.

File: src/core/SolarEclipseComputer.hpp
Comment (10110111) [reply src/core/SolarEclipseComputer.hpp]: OK, removed Nick and Alex.

**Comment by 10110111** [comment]
Let's wait until the CI finishes, after which, if successful, I'll merge.

File: src/core/SolarEclipseComputer.hpp
Comment (gzotti) [reply src/core/SolarEclipseComputer.hpp]: Sorry to be nitpicky, but now it looks strange. Worachate did most of this and should be first, I refactored a little bit (more possible).

Comment (10110111) [reply src/core/SolarEclipseComputer.hpp]: Is this OK? (Also removed "(Eclipses)" since the whole file is for eclipses now.)
```
 * Copyright (C) 2022 Worachate Boonplod
 * Copyright (C) 2022 Georg Zotti
 * Copyright (C) 2024 Ruslan Kabatsayev
```

**Pull request merged**
